packit/api.py

```diff
@@ -856,6 +856,84 @@ def sync_release(
             self.dg.local_project.git_repo.git.reset("--hard", "HEAD")
         return pr
 
+    def sync_push(
+        self,
+        source_git_branch: Optional[str] = None,
+        create_pr: bool = True,
+        title: Optional[str] = None,
+        description: Optional[str] = None,
+        force: bool = False,
+    ) -> Optional[PullRequest]:
+        """
+        When dist-git is updated then update the source-git repository by opening a PR.
+
+        Args:
+            source_git_branch: Branch in source-git, defaults to repo's default branch.
+            create_pr: Create a pull request if set to True.
+            force: Ignore changes in the git index.
+            title: Title (first line) of the commit & PR.
+            description: Description of the commit & PR.
+
+        Returns:
+            The created (or existing if one already exists) PullRequest if
+            create_pr is True, else None.
+
+        Raises:
+            PackitException, if the upstream repo or dist-git is dirty.
+        """
+        assert_existence(self.up.local_project, "Upstream local project")
+        assert_existence(self.dg.local_project, "Dist-git local project")
+
+        if self.up.is_dirty():
+            raise PackitException(
+                f"The upstream repository {self.up.local_project.working_dir} is dirty."
+                f"This is not supported."
+            )
+        if not force and self.dg.is_dirty():
+            raise PackitException(
+                f"The distgit repository {self.up.local_project.working_dir} is dirty,"
+                " will not discard the changes. Use --force to bypass."
+            )
+
+        pr = None
+        source_git_branch = source_git_branch or "main"
+        try:
+            self.up.checkout_branch(source_git_branch)
+            self.update_source_git()
+
+            if create_pr:
+                title = (
+                    title
+                    or f"Update to last dist-git commit: "
+                    f"{self.dg.local_project.git_repo.head.commit.message} "
+                    f"({self.dg.local_project.commit_hexsha})"
+                )
+                description = description or (
+                    f"Update upstream to last dist-git commit: "
+                    f"{self.dg.local_project.commit_hexsha}\n"
+                )
```

Comment:
Using the entire commit message in the title might be pretty long.

How about using the first line of the dist-git commit message as a default for title, and any other lines left as the default for `description`?

Mentioning the dist-git commit SHA  in the PR could be skipped, as it's already mentioned in the source-git commit message.

Reply:
I followed your suggestion. I have used the hexsha to create the PR title just in case there is no title str (the first line of the commit message is empty for some strange reason, even if I think this could not happen)